pkg/acceptance/barecluster/localcluster.go

```diff
@@ -138,61 +187,59 @@ func (c *Cluster) Close() {
 		n.Kill()
 	}
 	c.stopper.Stop(context.Background())
+	if c.cfg.Ephemeral {
+		_ = os.RemoveAll(c.cfg.DataDir)
+	}
 }
 
 // IPAddr returns the IP address of the specified node.
 func (c *Cluster) IPAddr(nodeIdx int) string {
-	if c.separateAddrs {
-		return fmt.Sprintf("127.1.1.%d", nodeIdx+1)
-	}
-	return "127.0.0.1"
+	return c.Nodes[nodeIdx].IPAddr()
 }
 
 // RPCAddr returns the RPC address of the specified node.
 func (c *Cluster) RPCAddr(nodeIdx int) string {
-	return fmt.Sprintf("%s:%d", c.IPAddr(nodeIdx), RPCPort(nodeIdx))
+	return net.JoinHostPort(c.IPAddr(nodeIdx), c.RPCPort(nodeIdx))
 }
 
-// RPCPort returns the RPC port of the specified node.
-func RPCPort(nodeIdx int) int {
-	return basePort + nodeIdx*2
+// RPCPort returns the RPC port of the specified node. Returns zero if unknown.
+func (c *Cluster) RPCPort(nodeIdx int) string {
+	return c.Nodes[nodeIdx].RPCPort()
 }
 
-// HTTPPort returns the HTTP port of the specified node.
-func HTTPPort(nodeIdx int) int {
-	return RPCPort(nodeIdx) + 1
+// HTTPPort returns the HTTP port of the specified node. Returns zero if unknown.
+func (c *Cluster) HTTPPort(nodeIdx int) string {
+	return c.Nodes[nodeIdx].HTTPPort()
 }
 
-func (c *Cluster) makeNode(nodeIdx int, binary string, extraArgs, extraEnv []string) *Node {
-	name := fmt.Sprintf("%d", nodeIdx+1)
-	dir := filepath.Join(dataDir, name)
-	logDir := filepath.Join(dir, "logs")
-	if err := os.MkdirAll(logDir, 0755); err != nil {
-		log.Fatal(context.Background(), err)
+func (c *Cluster) makeNode(wg *sync.WaitGroup, nodeIdx int, binary string, cfg NodeCfg) *Node {
+	node := &Node{
+		cfg: cfg,
 	}
 
 	args := []string{
 		binary,
 		"start",
 		"--insecure",
-		fmt.Sprintf("--host=%s", c.IPAddr(nodeIdx)),
-		fmt.Sprintf("--port=%d", RPCPort(nodeIdx)),
-		fmt.Sprintf("--http-port=%d", HTTPPort(nodeIdx)),
-		fmt.Sprintf("--store=%s", dir),
+		fmt.Sprintf("--host=%s", node.IPAddr()),
+		fmt.Sprintf("--port=%d", cfg.RPCPort),
+		fmt.Sprintf("--http-port=%d", cfg.HTTPPort),
+		fmt.Sprintf("--store=%s", cfg.DataDir),
+		fmt.Sprintf("--listening-url-file=%s", node.listeningURLFile()),
+		fmt.Sprintf("--pid-file=%s", node.pidFile()),
```

Comment:
It'll probably become obvious later, but why do you need `--pid-file`. Since we're exec'ing the process we'll know the pid, right?